### PR TITLE
feat(withLatestFrom): add pickLatestFrom and zipLatestFrom

### PR DESCRIPTION
**Pull request by briancavalier**
I figured I'd put a stake in the ground and see what happens ... maybe (as a friend of mine says) I'll end up falling on it, lol.  This is definitely POC quality and would need to be cleaned up before merging.  I'm hoping it helps with discussion, though.

This PR removes sample, in favor of adding pickLatestFrom and zipLatestFrom, where zipLatestFrom is basically sample with flipped stream arguments.

"Sample" has a clear connotation for continuous values, but not really for events: when an event is not occurring (which is the vast majority of the time), there is no value to sample.  "Sample" is used often in signal processing as analog -> digital.  There is no analog signal in this case, so, in reality, "sample" doesn't make much sense.  We were basically pretending to treat one of the event streams like an analog value.  I think everyone will be better off if we just treat it like what it is, and put some energy into considering how to integrate "real" analog values, maybe something like [this](https://github.com/briancavalier/most-behave) into the mostjs ecosystem.

### CODE REVIEW

File: packages/core/src/index.js.flow [hunk 1]
```diff
@@ -77,10 +77,13 @@ declare export function merge <A> (s1: Stream<A>, s2: Stream<A>): Stream<A>
 declare export function merge <A> (s1: Stream<A>): (s2: Stream<A>) => Stream<A>
 declare export function mergeArray <A> (ss: Array<Stream<A>>): Stream<A>
 
-declare export function sample <A, B, C> (f: (A, B) => C, sampler: Stream<A>, sampled: Stream<B>): Stream<C>
-declare export function sample <A, B, C> (f: (A, B) => C): (sampler: Stream<A>, sampled: Stream<B>) => Stream<C>
-declare export function sample <A, B, C> (f: (A, B) => C, sampler: Stream<A>): (sampled: Stream<B>) => Stream<C>
-declare export function sample <A, B, C> (f: (A, B) => C): (sampler: Stream<A>) => (sampled: Stream<B>) => Stream<C>
+declare export function withLatestFrom <A, B> (values: Stream<A>, stream: Stream<B>): Stream<A>
```
Comment (TylorS) [inline packages/core/src/index.js.flow]: I wouldn't bother with the second type parameter.
`withLatestFrom <A> (values: Stream<A>, stream: Stream<any>): Stream<A>`

File: packages/core/type-definitions/combinator/withLatestFrom.d.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+import { Stream } from '@most/types';
+
+export function withLatestFrom <A, B> (values: Stream<A>, s: Stream<B>): Stream<A>
```
Comment (TylorS) [inline packages/core/type-definitions/combinator/withLatestFrom.d.ts]: Same as the comment above

File: packages/core/src/index.js.flow (hunk 1 above)
Comment (briancavalier) [reply packages/core/src/index.js.flow]: Yeah, seems reasonable in this case :+1: ... I've just gotten in the habit of avoiding `any`

Comment (TrySound) [reply packages/core/src/index.js.flow]: Maybe better `*`?

**Comment by briancavalier** [comment]
Ah, I did flip the parameter order in this PR as we discussed, so I think its (note return type):

```js
// pickLatestFrom :: Stream a -> Stream b -> Stream a
const pickLatestFrom = (streamWhoseLatestValueWillBePicked: Stream<A>, streamWhoseValuesWillBeReplaced: Stream<B>): Stream<A> => ...
```

File: packages/core/src/index.js.flow (hunk 1 above)
Comment (briancavalier) [reply packages/core/src/index.js.flow]: Hmmm, I don't see any advantage to `*` over `any` in this particular case.  The type is completely ignored.  Do you see any advantage _in this particular case_?

Comment (TrySound) [reply packages/core/src/index.js.flow]: Just not using `any` :)

**Pull request closed without merging**